### DISCUSSION TITLE
Alternatives to time travel in >=2.3

### DISCUSSION THREAD

**Discussion by ifrancke-pryoninc** [milvus-io/milvus#38898] [Q&A and General discussion]
Hi. Since time travel search was officially deprecated in 2.3, I'm wondering if there are any official alternatives to the feature. Is there any other way to query an index at a certain time, or save a snapshot of the indexed data at a certain time?

**Comment by xiaofan-luan** [comment]
Thanks for the feedback. We are thinking of bring timetravel back, or some of it's alternatives, like Snapshot.

Can you explain your use case, like:
1. why you need this feature?
2. how often are you gonna to search historical data? how long do you want to keep it?
3. Is periodical snapshot gonna to work on your case?
4. Timetravel will keep many old data and can not be purged for a while, does this make sense to your use case?

**Comment by ifrancke-pryoninc** [comment]
The goal is to have a temporary backup of the data while other updates are run. We want the ability to rollback a change. Snapshot would be fine as well, but ideally we wouldn't have to store a backup externally. The timeline to restore the backup could be limited if necessary. Periodical would be perfectly fine.

**Comment by xiaofan-luan** [comment]
Sounds like perfect use case for snapshot.

Snapshot is a feature targeting at Milvus 3.0 (Est 2025 Q2)

Before snapshot is enabled u can periodically do backup
